### ISSUE TITLE
Invoice state update failed

### ISSUE THREAD

**Issue by interbiznw** [btcpayserver/btcpayserver#3214]
any idea why i get this issue? or the best place to look for debugging?

Steps to error, was go in invoices>select drop down for status, and try to change to invalid or settled, and get this error 

![image](https://user-images.githubusercontent.com/16450422/146640630-f912cd93-b552-4cf5-838d-fd1375f7ba3e.png)

**Comment by interbiznw** [comment]
got logging turned on and found this 

```2021-12-18 12:45:36.122 +00:00 [ERR] Connection id "0HME24FUFI6SP", Request id "0HME24FUFI6SP:00000001": An unhandled exception was thrown by the application.
System.Collections.Generic.KeyNotFoundException: The given key 'settled' was not present in the dictionary.```

**Comment by pavlenex** [comment]
Can you please some more details on your btcpay version, the enviroment/deployment method. I was not able to replicate this on 1.3.7.

**Comment by interbiznw** [comment]
BTCPayServer v1.3.7.0 (altcoins)

Built from source pulled earlier today on ubuntu


Everything else seems to be working, it does set the invoice status correct upon payment, I just cant seem to change it manually

**Comment by pavlenex** [comment]
Which invoice status are you trying to change? `new` to `invalid` then to `complete`, correct?

Unfortunately we don't provide support or debug to Altcoins build, as we don't really use it ourselves.

**Comment by interbiznw** [comment]
Yes it didn't matter what status.. but trying "new" to invalid or settled both result in the error. 

I was previously using the normal non altcoin build and it was doing the same thing

**Comment by pavlenex** [comment]
Interesting, could be something with the deployment method, maybe @NicolasDorier has some ideas on why this may happen on manual deployment.

**Comment by pavlenex** [comment]
@interbiznw were you able to figure this one out in the end?

**Comment by NicolasDorier** [comment]
Fixed by #3286 will be in next version.
